### PR TITLE
Upgrade minikube, kubectl, ubuntu, and go versions

### PR DESCRIPTION
**Pull request by kevinearls**
Signed-off-by: Kevin Earls <kearls@redhat.com>

### CODE REVIEW

File: .github/workflows/e2e-kubernetes.yaml [hunk 1]
```diff
@@ -3,17 +3,17 @@ on: [push, pull_request]
 
 jobs:
   end-to-end:
-    runs-on: ubuntu-16.04
+    runs-on: ubuntu-18.04
```
Comment (jpkrohling) [inline .github/workflows/e2e-kubernetes.yaml:6]: There's 20.04 available as well, IIRC.

Comment (kevinearls) [reply .github/workflows/e2e-kubernetes.yaml:6]: @jpkrohling I did try 20.04 and ran into the same problems.  This was mostly just an experiment to see if I could just change versions and have everything magically work.  Unfortunately that was not the case, and it looks like this upgrade will require at least a few more config changes.  I will probably pause this and get back to it when I have a few free hours.

Comment (jpkrohling) [reply .github/workflows/e2e-kubernetes.yaml:6]: If you get things working in minikube, you can probably get it working with kind, and I know kind does work on ubuntu-20.04. I think Linkerd uses it, and even had a KubeCon talk on the subject :)

**Comment by kevinearls** [comment]
This is now seriously out of date.

**Pull request closed without merging**